liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -228,32 +228,50 @@ public void stopServer() {
+        Set<Artifact> artifacts = project.getArtifacts();
+        List<String> artifactPaths = new ArrayList<String>();
+        for (Artifact artifact : artifacts) {
+            artifactPaths.add(artifact.getFile().getAbsolutePath());
         }
         
-        util = new DevMojoUtil(jvmOptions, serverDirectory);
-
+        // create an executor for tests with an additional queue of size 1, so any further changes detected mid-test will be in the following run
+        final ThreadPoolExecutor executor = new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS,
+                new ArrayBlockingQueue<Runnable>(1, true));
+        
+        log.info("Running maven-compiler-plugin:compile");
+        runMojo("org.apache.maven.plugins:maven-compiler-plugin", "compile", null, null);
+        log.info("Running maven-compiler-plugin:resources");
+        runMojo("org.apache.maven.plugins:maven-resources-plugin", "resources", null, null);
+        log.info("Running maven-compiler-plugin:testCompile");
+        runMojo("org.apache.maven.plugins:maven-compiler-plugin", "testCompile", null, null);
+        log.info("Running maven-compiler-plugin:testResources");
+        runMojo("org.apache.maven.plugins:maven-resources-plugin", "testResources", null, null);
         sourceDirectory = new File(sourceDirectoryString.trim());
         testSourceDirectory = new File(testSourceDirectoryString.trim());
+        
+        ArrayList<File> javaFiles = new ArrayList<File>();
+        listFiles(sourceDirectory, javaFiles, ".java");
+        
+        ArrayList<File> javaTestFiles = new ArrayList<File>();
+        listFiles(testSourceDirectory, javaTestFiles, ".java");
 
         log.debug("Source directory: " + sourceDirectory);
         log.debug("Output directory: " + outputDirectory);
         log.debug("Test Source directory: " + testSourceDirectory);
         log.debug("Test Output directory: " + testOutputDirectory);
 
-        // create an executor for tests with an additional queue of size 1, so any further changes detected mid-test will be in the following run
-        final ThreadPoolExecutor executor = new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS,
-                new ArrayBlockingQueue<Runnable>(1, true));
+        recompileJavaSource(javaFiles, artifactPaths, executor, true);
+        recompileJavaTest(javaTestFiles, artifactPaths, executor, true);
+
+        log.info("Running goal: create-server");
+        runMojo("net.wasdev.wlp.maven.plugins:liberty-maven-plugin", "create-server", serverName, null);
+        log.info("Running goal: install-feature");
+        runMojo("net.wasdev.wlp.maven.plugins:liberty-maven-plugin", "install-feature", serverName, null);
+        log.info("Running goal: install-apps");
+        runMojo("net.wasdev.wlp.maven.plugins:liberty-maven-plugin", "install-apps", serverName, null);
+        
+        util = new DevMojoUtil(jvmOptions, serverDirectory);
 
         util.addShutdownHook(executor);
 
```

Comment:
Looks like lines 278 to 286 can be removed since CreateServerMojo already does those when you call create-server goal in line 268.

Reply:
> Reply…

removed lines 278 to 286